ssl/ssl_test.go

```diff
@@ -138,20 +140,22 @@ func TestGenerateCerts(t *testing.T) {
 		}
 		secrets := &v1.Secret{Data: map[string][]byte{}}
 
-		err := GenerateCerts(certInfo, "namespace", "suffix", secrets)
+		err := GenerateCerts(certInfo, "namespace", "suffix", 700, secrets)
 
 		assert.NoError(t, err)
 		assert.NotEmpty(t, secrets.Data[certInfo[defaultCA].PrivateKeyName])
 		assert.NotEmpty(t, secrets.Data[certInfo[defaultCA].CertificateName])
 
 		certBlob, _ := pem.Decode(secrets.Data[certInfo[defaultCA].CertificateName])
-		if !assert.NotNil(t, certBlob, "Failed to decode certificate PEM block") {
-			return
-		}
+		require.NotNil(t, certBlob, "Failed to decode certificate PEM block")
+
 		cert, err := x509.ParseCertificate(certBlob.Bytes)
-		if assert.NoError(t, err) {
-			assert.Empty(t, cert.DNSNames, "CA cert should not include any DNS names")
-		}
+		require.NoError(t, err)
+
+		assert.True(t, cert.IsCA, "CA cert is a CA cert")
+		assert.True(t, cert.NotAfter.After(time.Now().Add(698*24*time.Hour)))
+		assert.True(t, cert.NotAfter.Before(time.Now().Add(702*24*time.Hour)))
```

Comment:
It might be a good idea to hold on to `time.Now()` in case this test runs _really_ slowly

Reply:
Seems like visual noise; the window is target time ± 2 days. If the test runs longer, it deserves to fail! 😄